src/deserialize/parse.rs

```diff
@@ -421,6 +454,17 @@ fn read_map(chars: &mut std::iter::Enumerate<std::str::Chars>) -> Result<Edn, Er
     }
 }
 
+fn read_comment(chars: &mut std::iter::Enumerate<std::str::Chars>) -> Result<Edn, Error> {
+    loop {
```

Comment:
we usually use `chars.take_while(|c| c != \n)`, instead of loops, as it is lazy. Also, have you thought about a test case when the comment doesnt end in a `\n`? I think there is a specific value that you can pass that encompasses new lines, eof, and returns.

Reply:
it handles EOF by matching None

I've considered files that use \r instead of \n, probably good to handle that as the spec doesn't emphatically say only \n line endings..

will also change to lazy version

Reply:
probably "and all subsequent characters to the next newline should be ignored" is meant to imply newline endings, otherwise it would need to be known whether to expect \r or \n or \r\n?

Reply:
Changed to lazy iterator expression.